### PR TITLE
Feature/inventory

### PR DESCRIPTION
**Pull request by BenRodProd**
Question about changing existing MongoDB values

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -48,16 +62,58 @@ export default function App({ Component, pageProps }) {
...
   }
-  const user = FetchUser();
+  useEffect(() => {
+    if (firstLoad) {
+      router.push("/library");
+    }
+  }, []);
+  useEffect(() => {
+    if (Array.isArray(data)) {
+      if (Array.isArray(user)) {
```
Comment (ThomasKirst) [inline pages/_app.js:107]:
```suggestion
    if (Array.isArray(data) && Array.isArray(user)) {
```

File: pages/_app.js [hunk 2]
```diff
@@ -86,8 +142,24 @@ export default function App({ Component, pageProps }) {
...
+          {firstLoad ? (
+            <TitleScreen
+              priority
+              src="/assets/MINDBLADE.png"
+              alt="TitleScreen"
+              onClick={() => setFirstLoad(false)}
+              height="1920"
+              width="1080"
+            ></TitleScreen>
+          ) : null}
```
Comment (ThomasKirst) [inline pages/_app.js:156]:
```suggestion
          {firstLoad && 
            <TitleScreen
              priority
              src="/assets/MINDBLADE.png"
              alt="TitleScreen"
              onClick={() => setFirstLoad(false)}
              height="1920"
              width="1080"
            />
          }
```

File: pages/api/users/index.js [hunk 3]
```diff
@@ -9,4 +9,9 @@ export default async function handler(request, response) {
 
     return response.status(200).json(users);
   }
+  if (request.method === "PUT") {
+    await User.find();
+
+    response.status(200).json({ status: `User ${id} updated!` });
+  }
```
Comment (ThomasKirst) [inline pages/api/users/index.js:16]: This has no purpose, I guess. You should remove it.

**Review by ThomasKirst — CHANGES_REQUESTED** [review]
That was an epic odyssey. You left some surprising treasures to discover 😃

Overall a very huge PR with a complex data structure which makes it challenging to wrap one´s head around it.

In general there are a few optimisation opportunities which I see here:
- Some of the code could be extracted into functions.
- At some point it seemed as if you could use "derived" state from existing objects instead of creating separate states.

Great work overall, some minor issues detected where you should reflect on.

Keep it up 👍

File: components/EnterLibrary.js [hunk 4]
```diff
@@ -13,19 +29,38 @@ const StyledImage = styled(Image)`
...
+  const router = useRouter();
+  function handleLeaveLibrary() {
+    if (
+      !currentBook ||
+      currentBook === "" ||
+      userData[0].currentBook !== currentBook
+    ) {
+      setPopupActive(true);
+      setTimeout(() => setPopupActive(false), 3000);
+      router.push("/library");
```
[the remark this answers is not in the document]
Comment (BenRodProd) [reply components/EnterLibrary.js:50]: this is when no currentbook is chosen (deleted book and went back to main game)... but the router.push ist not necessary there, you're right

File: pages/index.js [hunk 5]
```diff
@@ -9,22 +9,35 @@ import LevelBackgroundImage from "@/components/LevelBackgroundImage";
...
+    userData[0].books[
+      userData[0].books.findIndex((element) => element.bookname === currentBook)
+    ].inventory
+  );
+  const [inventorySlots, setInventorySlots] = useState(
+    userData[0].books[0].inventorySlots
+  );
+  const [activeBookLibrary, setActiveBookLibrary] = useState(
+    library.filter((element) => element.book === currentBook)
+  );
```
[the remark this answers is not in the document]
Comment (BenRodProd) [reply pages/index.js:40]: Oops, is not used at all... Tried to begin with the new logic of choosing questions and only select questions of the user. Will do this tomorrow, after the session about next/auth...

**Pull request merged**